src/_common/report/form/form.ts

```diff
@@ -1,22 +1,309 @@
+		if (!this.formModel.reason) {
+			return true;
+		}
+		const reason = this.reasons.find(i => i.radioValue === this.formModel.reason);
+		if (!reason) {
+			return true;
+		}
+		if (reason.contexts && reason.contexts.length > 0) {
+			if (!this.formModel.context || this.formModel.context.length === 0) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	get isDescriptionOptional() {
+		if (!this.formModel.reason || !this.formModel.context) {
+			return true;
+		}
+
+		return !(this.formModel.context.length > 0 && this.formModel.context.includes('other'));
+	}
+
+	get reasons(): any[] {
+		switch (this.type) {
+			case 'Game':
+				return [
+					{
+						radioValue: 'spam',
+						text: this.$gettext('It is spam or unwanted commercial content'),
+					},
+					{
+						radioValue: 'abuse',
+						text: this.$gettext(
+							'Incorrect maturity rating for the content in the game'
+						),
+					},
+					{
+						radioValue: 'stolen',
+						text: this.$gettext('Game does not belong to this developer'),
+						source: {
+							placeholder: this.$gettext(
+								/** TODO(copy) */ 'Tell us where this game was stolen from'
+							),
+						},
+					},
+					{
+						radioValue: 'no-info',
+						text: this.$gettext(
+							'No information on game page (no screenshots, sparse description, placeholder page, etc.)'
+						),
+					},
+					{
+						radioValue: 'malware',
+						text: this.$gettext('Virus or other form of malware'),
+						infoText: this.$gettext(
+							/** TODO(copy) */ "Some text about that just because that game doesn't start or their antivirus detects it, it does not have to be a virus."
+						),
+					},
+				];
+			case 'Fireside_Post':
+				const reasons = [
+					{
+						radioValue: 'spam',
+						text: this.$gettext('It is spam or unwanted commercial content'),
+					},
+					{
+						radioValue: 'abuse',
+						text: this.$gettext('It is hate speech or contains graphic content'),
+					},
+					{
+						radioValue: 'explicit',
+						text: this.$gettext(
+							'It is pornographic or contains sexually explicit material'
+						),
+					},
+					{
+						radioValue: 'harassment',
+						text: this.$gettext('It is harassment or bullying'),
+					},
+					{
+						radioValue: 'stolen',
+						text: this.$gettext('Content in this post does not belong to the author'),
+						source: {
+							placeholder: this.$gettext(
+								/** TODO(copy) */ 'Tell us where content in this post was stolen from'
+							),
+						},
+					},
+				];
+
+				// For a devlog post of a game that is maturity restricted, we don't want to show the "explicit" report option.
+				// Those devlog posts can be explicit, and we don't want to encourage false reports.
+				if (
+					this.resource instanceof FiresidePost &&
+					this.resource.game instanceof Game &&
+					this.resource.game.tigrs_age === 3
+				) {
+					arrayRemove(reasons, i => i.radioValue === 'explicit');
+				}
```

Comment:
you're assuming game posts only show on the game page.
this might be changing very soon as you know (wink wink).
consider reverting this.

Reply:
I'd argue if we allow showing the post somewhere else, we can add that check here.
Only show this option, if the post is solely visible on the game page.